src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2058,7 +2068,20 @@ def aks_runcommand(cmd, client, resource_group_name, name, command_string="", co
         command_id = command_id_regex.findall(command_result_polling_url)[0]
         _aks_command_result_in_progess_helper(client, resource_group_name, name, command_id)
         return
-    return _print_command_result(cmd.cli_ctx, command_result_poller.result(300))
+
+    progress_bar = PollerProgressBar(cmd.cli_ctx, command_result_poller)
+    progress_bar.begin()
+
+    now = datetime.datetime.now()
+    while not command_result_poller.done():
+        if datetime.datetime.now() - now >= datetime.timedelta(seconds=300):
+            break
+
+        progress_bar.update_progress()
+        time.sleep(0.5)
+
+    progress_bar.end()
+    return _print_command_result(cmd.cli_ctx, command_result_poller.result())
```

Comment:
so if the operation is not completed within 300 seconds, the progress bar will end, but the polling here will continue (until it times out on the server side?)

Reply:
From what I understand, if the operation is not completed in 300s, the failure message will be logged and the python process (including the poller) will be terminated.

Reply:
I did a simple test based on your branch, the following command is executed and does not terminate after 5 minutes. After 5 minutes, only the progress bar is no longer displayed, and the command finally returns in 10 minutes.

`az aks command invoke -g xxx -n xxx -c "sleep 600 && echo hello"`

Reply:
Fixed, I've added a timeout to the LRO poller result. 
```go
    return _print_command_result(cmd.cli_ctx, command_result_poller.result(timeout=0))
```